armi/reactor/components/component.py

```diff
@@ -906,8 +943,11 @@ def getMass(self, nuclideNames=None):
         )
         nuclideNames = self._getNuclidesFromSpecifier(nuclideNames)
         # densities comes from self.p.numberDensities
-        densities = self.getNuclideNumberDensities(nuclideNames)
-        nDens = {nuc: dens for nuc, dens in zip(nuclideNames, densities)}
+        if len(nuclideNames) > 0:
+            densities = self.getNuclideNumberDensities(nuclideNames)
+            nDens = {nuc: dens for nuc, dens in zip(nuclideNames, densities)}
```

Comment:
`dict` can take in a `zip`
```python
>>> k = ["u", "b"]
>>> d = [1, 2]
>>> dict(zip(k, d))
{'u': 1, 'b': 2}
```
```suggestion
            nDens = dict(zip(nuclideNames, densities))
```